studio/app/optinist/wrappers/caiman/__init__.py

```diff
@@ -11,5 +12,9 @@
             "function": caiman_cnmf,
```

Comment:
@quanpython 
When executing caiman, the following error occurred in some cases.
Can you please investigate the situation?

- Procedure
  - RUNN ALL caiman_mc -> caiman_cnmf
    - Note: Immediately after the error occurred, RUNN ALL again, and the process ended successfully.

- error log

> 2023-09-22 16:00:09,362  INFO:     127.0.0.1:51450 - "POST /run/result/default/527aeb81 HTTP/1.1" 500
> 2023-09-22 16:00:09,363  ERROR:    Exception in ASGI application
> Traceback (most recent call last):
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
>     result = await app(  # type: ignore[func-returns-value]
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
>     return await self.app(scope, receive, send)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/fastapi/applications.py", line 290, in __call__
>     await super().__call__(scope, receive, send)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/applications.py", line 122, in __call__
>     await self.middleware_stack(scope, receive, send)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/middleware/errors.py", line 184, in __call__
>     raise exc
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/middleware/errors.py", line 162, in __call__
>     await self.app(scope, receive, _send)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/middleware/cors.py", line 91, in __call__
>     await self.simple_response(scope, receive, send, request_headers=headers)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/middleware/cors.py", line 146, in simple_response
>     await self.app(scope, receive, send)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
>     raise exc
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
>     await self.app(scope, receive, sender)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
>     raise e
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
>     await self.app(scope, receive, send)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/routing.py", line 718, in __call__
>     await route.handle(scope, receive, send)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/routing.py", line 276, in handle
>     await self.app(scope, receive, send)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/starlette/routing.py", line 66, in app
>     response = await func(request)
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/fastapi/routing.py", line 241, in app
>     raw_response = await run_endpoint_function(
>   File "/Users/user1/mambaforge-pypy3/envs/optinist_pyproject/lib/python3.8/site-packages/fastapi/routing.py", line 167, in run_endpoint_function return await dependant.call(**values)
>   File "/workspace/barebone-studio/studio/app/common/routers/run.py", line 33, in run_result
>     return WorkflowResult(workspace_id, uid).get(nodeDict.pendingNodeIdList)
>   File "/workspace/barebone-studio/studio/app/common/core/workflow/workflow_result.py", line 49, in get
>     results[node_id] = NodeResult(
>   File "/workspace/barebone-studio/studio/app/common/core/workflow/workflow_result.py", line 95, in __init__
>     self.info = PickleReader.read(pickle_filepath)
>   File "/workspace/barebone-studio/studio/app/common/core/utils/pickle_handler.py", line 13, in read
>     return pickle.load(f)
> EOFError: Ran out of input
>

Reply:
@itutu-tienday
The above error does not appear on my environment. Do you run the workflow with default parameters?

With the information you provided, I determined that the above error occurred when calling the POST run/result API, the output is taken from the pickle file, but this pickle file has no data leading to the error `EOFError: Ran out of input`

Because the above error log is an API log, not a workflow log, I cannot determine what happened during processing.

Could you please provide more information, e.g. error.log file or workflow parameters, etc...

Reply:
@quanpython 
Sorry I don't seem to have enough information.
I will add some information.

- The target experiment.yaml is attached.
  - [experiment.yaml.zip](https://github.com/arayabrain/barebone-studio/files/12723841/experiment.yaml.zip)
  - The error log above is for the API, but the workflow also seemed to stop running. (success: running)
  - However, there seemed to be no error log from smakemake. (error.log is empty)
- Also, there seemed to be no empty pkl files in the output folder of the target workflow.

That's all for now.
If it is difficult to investigate and the reproducibility is low, I would like to put this matter on hold for now.

Reply:
close temporarily